### DISCUSSION TITLE
Brute Force

### DISCUSSION THREAD

**Discussion by quer0016** [JanssenProject/jans#7755] [Q&A] [answered]
Hello, do you know if there is a way to protect from brute force attacks? Right now I find the solution using fail2ban but I would like not to use third-party software.

**Comment by nynymike** [comment, marked as answer]
This could be handled in a custom authn interception script or an Agama Flow by writing login metadata to the cache. We have such a script for oxauth: https://github.com/GluuFederation/oxAuth/tree/master/Server/integrations/basic.lock.account 

We plan to add this feature into the Gluu [agama-pw](https://github.com/GluuFederation/oxAuth/tree/master/Server/integrations/basic.lock.account) project

**Reply by quer0016** [reply]
thx Mike.
